src/utils/httpClient.ts

```diff
@@ -0,0 +1,36 @@
+import axios from 'axios'
+
+export type AxiosDNSLookupFunction = typeof axios.defaults.lookup
+
+export const httpClient = axios.create({})
+
+export const sendAxiosRequestWithRequestInit = async (url: string | URL, requestInit: RequestInit) => {
```

Comment:
We currently use the [`RequestInit`](https://fetch.spec.whatwg.org/#requestinit) type from the Fetch API for request options. However, we may need to switch this to [`AxiosRequestConfig`](https://github.com/axios/axios#request-config). In the current implementation, the `sendAxiosRequestWithRequestInit` method allows properties like `keepalive`, `bodyInit`, `cache`, `credentials`, `integrity`, `mode`, `priority`, `redirect`, `referrer`, `referrerPolicy`, `signal`, and `window` to be specified in requestInit. However, these properties are not meaningful for `axios.request` and are therefore unused. As a result, the `...rest` part is also effectless. To address this, we could consider changing the type to `AxiosRequestConfig` or, if we want to allow specific customizations, we could use a partial type of it.

This change impacts all test cases where `RequestInit` is used, which might require some modifications. I'm also okay with addressing it in a separate PR.

Reply:
I agree that we need to use `AxiosRequestConfig` here